### PR TITLE
Clippy fixes.

### PR DESCRIPTION
**Pull request by Narsil**
Fixing all new clippy warnings.

### CODE REVIEW

File: bindings/node/native/src/tasks/tokenizer.rs [hunk 1]
```diff
@@ -11,7 +11,7 @@ pub enum EncodeTask<'s> {
 }
 
 pub enum EncodeOutput {
-    Single(Encoding),
+    Single(Box<Encoding>),
```
Comment (McPatate) [inline bindings/node/native/src/tasks/tokenizer.rs:14]: Is `Box` needed because Javascript? 😅

File: bindings/node/package.json [hunk 2]
```diff
@@ -29,7 +29,7 @@
     "eslint-plugin-prettier": "^3.4.1",
     "eslint-plugin-simple-import-sort": "^5.0.3",
     "jest": "^26.6.3",
-    "neon-cli": "^0.3.3",
+    "neon-cli": "^0.9.1",
```
Comment (McPatate) [inline bindings/node/package.json:32]: Big jump! 😁

**Review by McPatate — APPROVED** [review]
LGTM

File: bindings/node/native/src/tasks/tokenizer.rs (hunk 1 above)
Comment (Narsil) [reply bindings/node/native/src/tasks/tokenizer.rs:14]: No, clippy yells becuse that variant is 240 bytes long (pretty long for an enum). Setting it as a box just uses a pointer instead.

File: bindings/node/package.json (hunk 2 above)
Comment (Narsil) [reply bindings/node/package.json:32]: `npm audit --fix` did it !

**Comment by Narsil** [comment]
I will merge this so we can rebase other PRs.

@n1t0 we can still discuss some choices here.

- 3.6 MacOS as a target was dropped by GA (so we cannot run it)
- Had to modify the testing because MacOS Python 3.8+ uses `spawn` and not `fork` anymore. (Just modifying the tests seems reasonable)
- Had to bump `setup-python@v2` because otherwise the lib location was not found by rust.
- Had to `npm audit --fix` (security warning for `tmpl`) 
- `pyarrow < 3.0` was invalid since now datasets requires `>= 3.0` and the original reason for downgrading seems gone.

**Pull request merged**